engine/class_modules/sc_death_knight.cpp

```diff
@@ -7057,10 +7070,10 @@ void death_knight_t::trigger_soul_reaper_death( player_t* target )
 
   if ( td -> dot.soul_reaper -> is_ticking() )
   {
-    sim -> print_log( "Target {} died while affected by Soul Reaper, player {} gains Soul Reaper buff.",
+    sim -> print_log( "Target {} died while affected by Soul Reaper, player {} gains Runic Corruption buff.",
                       target -> name(), name() );
 
-    buffs.soul_reaper -> trigger();
+    buffs.runic_corruption -> trigger();
```

Comment:
The dk module uses death_knight_t::trigger_runic_corruption( double rpcost, double override_chance, proc_t* proc ) as a helper function to trigger the runic corruption, this is not only useful for RP spenders, but also to track the different sources of Runic Corruption (similarly to trigger_killing_machine() or trigger_festering_wound()). 

You'll have to create and init a new proc for runic corruption generated by Soul Reaper and send it to trigger_runic_corruption(), the function will handle the rest.